### ISSUE TITLE
indentation reports a false error on LESS multiline mixin usage

### ISSUE THREAD

**Issue by diagramatics** [stylelint/stylelint#1149]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

The rule `indentation` has a bug where mixin calls that act as a blockless rule (?) is giving false indentation errors. Not sure if the PoC thing @jeddy3 mentioned would solve this but I did test implementing the test case in the latest `master` branch and it still does the same thing. I was about to fix this myself and make a PR but I don't think I have time to sort this out 😞 

Sidenote, maybe we can use the ES6 template tags for the test cases codes? Anyway, here are the formalities.

> Which rule, if any, is this issue related to?

`indentation`.

> What CSS is needed to reproduce this issue?

```less
.foo {
    .mixin(
        @foo,
        @bar,
        @baz
    );
}
```

> What stylelint configuration is needed to reproduce this issue?

Particularly in my setup is:
```json
"indentation": [4, {
  "indentInsideParens": "once"
}]
```

> Which version of stylelint are you using?

`6.2.2` on my codebase, but tested on the master branch with a test case and it throws an error.

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS with Gulp.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Related to `postcss-less`.

> What did you expect to happen?

No lint errors.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
3:9	⚠  Expected indentation of 4 spaces (indentation) [stylelint]
4:9	⚠  Expected indentation of 4 spaces (indentation) [stylelint]
5:9	⚠  Expected indentation of 4 spaces (indentation) [stylelint]
6:5	⚠  Expected indentation of 0 spaces (indentation) [stylelint]
```

**Comment by jeddy3** [comment]
> Not sure if the PoC thing @jeddy3 mentioned would solve this but I did test implementing the test case in the latest master branch and it still does the same thing.

Thanks for testing against `master`. Unfortunately, my PoC won't have helped here as those particular utilities aren't used within `indentation`. I'm not sure off-hand if they are applicable though. Feel free to investigate though.

**Comment by diagramatics** [comment]
So I've done some investigating, and I think this issue can be handled the same way as my last LESS related issue #1077 and with @jeddy3's #1128 in `isStandardSelector`, or handle it as an `atRule` with the same logic as `checkAtRuleParams`.

Essentially with `checkValue` and `checkAtRuleParams` the indentation level of the parameters on the next newlines are raised by 1. LESS mixins don't get the treatment because they are rules and rules don't get their expected indentation level raised by 1.

**Comment by diagramatics** [comment]
Interestingly, I've tested this code:
```
.test {
  .mixin-test(
    @a: true
  );
  
  color: rgb(
    255,
    255,
    255
  );
}
```

To test both a LESS mixin usage and a multiline declaration without `indentInsideParams: once` and the `color` declaration doesn't spit out errors, as well as the mixin usage which is expected. Have I uncovered a bug on `indentInsideParams` in the process?

**Comment by Nirazul** [comment]
I'm not sure on what config and rules you've tested in your example. Maybe you could clarify this by adding this info to your post.

Concerning declaring indentation without `indentInsideParams` I've found this on the stylelint page:
> By default, indentation within function arguments and other parentheses are ignored. If you would like to enforce indentation inside parentheses, use this option.

So everything inside functions will be ignored.

**Comment by jeddy3** [comment]
> So everything inside functions will be ignored.

That's right. You need to turn of checking of parens with the `indentInsideParens` option.

> Essentially with checkValue and checkAtRuleParams in indentation/index.js the indentation level of the parameters on the next newlines are raised by 1. LESS mixins don't get the treatment because they are rules and rules don't get their expected indentation level raised by 1.

That's some good digging and investigating! Thanks :)

Feel free to contribute a fix.

**Comment by ntwb** [comment]
Related: #1451, #1458

**Comment by davidtheclark** [comment]
To follow up on one comment you made above, @diagramatics: "maybe we can use the ES6 template tags for the test cases codes" —— we can unless the rule is sensitive to indentation.

**Comment by diagramatics** [comment]
Regarding the ES6 template tags thing, you can always use the `stripIndent` tag from https://github.com/declandewet/common-tags.

**Comment by davidtheclark** [comment]
@diagramatics Good to know about. PR implementing it would be awesome. Otherwise we'll keep it in mind.
